# HG changeset patch
# User Rik <rik@octave.org>
# Date 1503706227 25200
#      Fri Aug 25 17:10:27 2017 -0700
# Node ID d837f7f6e4aa49e83fb64e3960621a99617f0628
# Parent  c7b801f36be4dbcbfc7853e1862d1574a1d80fd3
Don't label all values on bar plot when there are more than 15 bars (bug #51773).

* __bar__.m: Check that the number of bar elements to label is <= 15 before
labeling them; Otherwise, use default selected tick positions.

diff --git a/scripts/plot/draw/private/__bar__.m b/scripts/plot/draw/private/__bar__.m
--- a/scripts/plot/draw/private/__bar__.m
+++ b/scripts/plot/draw/private/__bar__.m
@@ -195,17 +195,19 @@ function varargout = __bar__ (vertical, 
     endif
     unwind_protect
       hax = newplot (hax);
 
       htmp = bars (hax, vertical, x, y, xb, yb, gwidth, group,
                    have_line_spec, bv, newargs{:});
 
       if (! ishold ())
-        if (all (x(:,1) == fix (x(:,1))))
+        if (numel (x(:,1)) <= 15 && all (x(:,1) == fix (x(:,1))))
+          ## Set manual ticks, rather than relying on autoselection,
+          ## when ticks are a small number of integers.
           if (vertical)
             set (hax, "xtick", x(:,1));
           else
             set (hax, "ytick", x(:,1));
           endif
         endif
         ## Hack prevents color and xlim setting changes when basevalue changes.
         if (vertical)
